wherehows-frontend/app/security/AuthenticationManager.java

```diff
@@ -13,174 +13,59 @@
  */
 package security;
 
-import dao.UserDAO;
-import javax.naming.directory.SearchControls;
-import wherehows.models.table.User;
-import org.apache.commons.lang3.StringUtils;
-import play.Logger;
-import play.Play;
-
-import java.util.Hashtable;
-import java.util.Map;
-import java.util.Collections;
-import java.util.HashMap;
+import javax.security.auth.callback.NameCallback;
+import javax.security.auth.callback.PasswordCallback;
+import javax.security.auth.callback.UnsupportedCallbackException;
 
 import javax.naming.AuthenticationException;
-import javax.naming.CommunicationException;
-import javax.naming.Context;
 import javax.naming.NamingException;
-import javax.naming.NamingEnumeration;
-import javax.naming.directory.Attribute;
-import javax.naming.directory.DirContext;
-import javax.naming.directory.InitialDirContext;
-import javax.naming.directory.SearchResult;
-
+import javax.security.auth.login.LoginContext;
+import javax.security.auth.login.LoginException;
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.auth.callback.Callback;
 
 public class AuthenticationManager {
 
-  private static final String MASTER_LDAP_URL_KEY = "authentication.ldap.url";
-  private static final String MASTER_PRINCIPAL_DOMAIN_KEY = "authentication.principal.domain";
-  private static final String LDAP_CONTEXT_FACTORY_CLASS_KEY = "authentication.ldap.context_factory_class";
-  private static final String LDAP_SEARCH_BASE_KEY = "authentication.ldap.search.base";
-
-  private static final String LDAP_DISPLAY_NAME_KEY = "displayName";
-  private static final String LDAP_MAIL_KEY = "mail";
-  private static final String LDAP_DEPARTMENT_NUMBER_KEY = "departmentNumber";
-
-  private static final String LDAP_USER_ATTR_FILTER_EXPRESSION = "(&(objectClass=user)(sAMAccountName={0}))";
-
-  private static final String contextFactories =
-      Play.application().configuration().getString(LDAP_CONTEXT_FACTORY_CLASS_KEY);
-  /*  three LDAP properties, each is a '|' separated string of same number of tokens. e.g.
-      Url: "ldaps://ldap1.abc.com:1234|ldap://ldap2.abc.com:5678"
-      Principal Domain: "@abc.com|@abc.cn"
-      Search Base: "ou=Staff Users,dc=abc,dc=com|ou=Staff Users,dc=abc,dc=cn"
-   */
-  private static final String[] ldapUrls =
-      Play.application().configuration().getString(MASTER_LDAP_URL_KEY).split("\\s*\\|\\s*");
-  private static final String[] principalDomains =
-      Play.application().configuration().getString(MASTER_PRINCIPAL_DOMAIN_KEY).split("\\s*\\|\\s*");
-  private static final String[] ldapSearchBase =
-      Play.application().configuration().getString(LDAP_SEARCH_BASE_KEY).split("\\s*\\|\\s*");
-
   public static void authenticateUser(String userName, String password) throws NamingException {
     if (userName == null || userName.isEmpty() || password == null || password.isEmpty()) {
       throw new IllegalArgumentException("Username and password can not be blank.");
```

Comment:
Let's use Guava's Preconditions for arg check: https://github.com/google/guava/wiki/PreconditionsExplained